program/rust/src/lib.rs

```diff
@@ -33,11 +36,34 @@ pub extern "C" fn c_entrypoint(input: *mut u8) -> u64 {
 
 #[no_mangle]
 pub extern "C" fn entrypoint(input: *mut u8) -> u64 {
-    let c_ret_val = unsafe { c_entrypoint(input) };
-    if c_ret_val == c_oracle_header::SUCCESSFULLY_UPDATED_AGGREGATE {
-        //0 is the SUCCESS value for solana
-        return 0;
-    } else {
-        return c_ret_val;
+    let (program_id, accounts, instruction_data) = unsafe { deserialize(input) };
+    let cmd_hdr_size = ::std::mem::size_of::<c_oracle_header::cmd_hdr>();
+    if instruction_data.len() < cmd_hdr_size {
+        panic!("insufficient data, could not parse instruction");
+    }
```

Comment:
Instead of panicking can we a more conventional Solana error handling. Checkout the logging branch so see how to.